src/main/java/org/jfree/chart/entity/StandardEntityCollection.java

```diff
@@ -139,6 +140,28 @@ public ChartEntity getEntity(double x, double y) {
         return null;
     }
 
+    /**
+     * Returns a collection of entities whose area contain the specified point
+     *
+     * @param x the x coordinate.
+     * @param y the y coordinate.
+     *
+     * @return A Vector of ChartEntity objects, possibly empty, never null
+     */
+    @Override
+    public Vector<ChartEntity> getEntities(double x, double y){
+        Vector<ChartEntity> filtered = new Vector<>();
+
+        int entityCount = this.entities.size();

```

Comment:
Maybe the Stream API here to filter the list?

Reply:
Oh - I forgot that was an option lol.  When I initially made the changes, we were still on Java 7, so it didn't even cross my mind.  We had to update to Java 8, anyway, though (for a really stupid reason, actually. Don't get me started hahaha!), and by then I had forgotten all about this change!